baton/hub.go

```diff
@@ -40,45 +40,57 @@ func (h hub) run() {
 				close(c.send)
 			}
 		case rawMsg := <-h.receive:
-			fmt.Println(rawMsg)
 			var cmd command
 			if err := json.Unmarshal(rawMsg.data, &cmd); err != nil {
 				fmt.Println("nu")
 				log.Println(err)
 				break
 			}
-
+			fmt.Println(cmd)
 			msg := msg{rawMsg.conn, cmd}
+			processMessage(msg)
+		}
+	}
+}
 
-			fmt.Println(cmd)
-			resp := make(chan interface{})
-			module, ok := modules[msg.cmd.Module]
+func processMessage(message msg) {
+	resp := make(chan interface{})
+	module, ok := modules[message.cmd.Module]
+	if !ok {
+		fmt.Println(message.cmd.Module, "not in", modules)
+		return
+	}
+
+	go func() {
+		for {
+			r, ok := <-resp
 			if !ok {
-				fmt.Println(msg.cmd.Module, "not in", modules)
+				close(resp)
 				break
 			}
-
-			go func() {
-				for {
-					r, ok := <-resp
-					if !ok {
-						close(resp)
-					}
-					bytes, err := json.Marshal(command{cmd.Module, cmd.Call, r}) //add in Module and Call info for client
-					if err != nil {
-						log.Println("Error Marshaling")
-						log.Println(err)
-						break
-					}
-					msg.conn.send <- bytes
-				}
-			}()
-
-			if err := module.RunCommand(cmd.Call, cmd.Body, resp); err != nil {
-				fmt.Println(err)
-				fmt.Println("the fuck happened to this command?")
+			bytes, err := json.Marshal(command{message.cmd.Module, message.cmd.Call, r}) //add in Module and Call info for client
+			if err != nil {
+				log.Println("Error Marshaling")
+				log.Println(err)
 				break
 			}
+			message.conn.send <- bytes
 		}
-	}
+	}()
```

Comment:
A new goroutine will be created but will not run infinitely unless the command never ends.